### PR TITLE
[iOS] Fixed NRE in SearchBarRenderer OnSearchButtonClicked

### LINKED ISSUE #11865: iOS SearchBarRenderer throws a NullReferenceException when the SearchButton redirects to a new page
### Description
In cases where the iOS SearchBar OnSearchButtonClicked event handler redirects to a different page, the SearchBarRenderer throws a NullReferenceException.
This only happens on iOS.

### Steps to Reproduce

1. Launch provided app
2. Type some text into the search bar
3. Press the "Search" button

### Expected Behavior
When navigating to a new page from within an OnSearchButtonClicked event handler, no exceptions are thrown.

### Actual Behavior
The application crashes when navigating to a new page from within an OnSearchButtonClicked event handler.

### Basic Information

- Version with issue: 4.8.0.1269
- Last known good version: Unknown
- IDE: Visual Studio Enterprise 2019 16.6.5
- Platform Target Frameworks:
  - iOS:  13.20.2.2
- Nuget Packages:
  - NETStandard.Library v2.0.3
  - Xamarin.Forms v4.8.0.1269
- Affected Devices: Suspected all, but tested on iPhone 6, iPhone SE, and iPad Pro

### Reproduction Link

[SearchBarRendererExceptionDemo.zip](https://github.com/xamarin/Xamarin.Forms/files/5105143/SearchBarRendererExceptionDemo.zip)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Fixed NRE in SearchBarRenderer OnSearchButtonClicked on iOS.

### Issues Resolved ### 

- fixes #11865 

### API Changes ###

 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 
fix11865

### Testing Procedure ###
Launch Core Gallery and navigate to the issue 11865. Type something in the SearchBar and tap the search button. If can navigate without any crash (disposed Control), the test has passed.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/SearchBarRenderer.cs [hunk 1]
```diff
@@ -216,7 +216,7 @@ void OnEditingStarted(object sender, EventArgs e)
 		void OnSearchButtonClicked(object sender, EventArgs e)
 		{
 			Element.OnSearchButtonPressed();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/SearchBarRenderer.cs:218]:
```suggestion
			Elemen?t.OnSearchButtonPressed();
```

Comment (rmarinho) [reply Xamarin.Forms.Platform.iOS/Renderers/SearchBarRenderer.cs:218]: Just in case

**Review by rmarinho — CHANGES_REQUESTED** [review]

File: Xamarin.Forms.Platform.iOS/Renderers/SearchBarRenderer.cs (hunk 1 above)
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/SearchBarRenderer.cs:218]:
```suggestion
			Element?.OnSearchButtonPressed();
```

Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.iOS/Renderers/SearchBarRenderer.cs]: Applied the changes.

**Review by rmarinho — APPROVED** [review]

**Pull request merged**